### PR TITLE
Update sidebar.blade.php

### PR DESCRIPTION
**Pull request by joelmellon**
The current sidebar menu implementation isn't very smart when it comes to detecting which menu item should be "active" - it merely looks for the beginning of the URL in the href (and vice versa) which leads to multiple menu items being highlighted at once when they both begin with the same word, eg. `user` and `usersetting`. See screenshot for my use case:

Screen Shot 2020-08-15 at 10 58 52 AM
Above, I've got CRUD for models called `trail` and `trailhead` which means that when I'm on the trailhead page, both the Trailheads menu item is active (correctly) and the Trails menu item is active (incorrectly).

This pull request implements improved logic to determine if menu item is active.

It could be made in better by also comparing the URL parameters (URL segment after the `?`) but I think this is a significant improvement already.

### CODE REVIEW

File: src/resources/views/base/inc/sidebar.blade.php [hunk 1]
```diff
@@ -58,11 +58,11 @@ function() { return $(this).attr('href') === full_url; }
       // If not found, look for the link that starts with the url
       if(!$curentPageLink.length > 0){
           $curentPageLink = $navLinks.filter( function() {
-            if ($(this).attr('href').startsWith(full_url)) {
+            if ($(this).attr('href').split('?')[0] === full_url) {
```
Comment (joelmellon) [inline src/resources/views/base/inc/sidebar.blade.php:61]: JavaScript's split always returns an array with at least one value, so even if there's no `?` this still works as expected.

File: src/resources/views/base/inc/sidebar.blade.php [hunk 2]
```diff
@@ -58,11 +58,11 @@ function() { return $(this).attr('href') === full_url; }
...
               return true;
             }
 
-            if (full_url.startsWith($(this).attr('href'))) {
+            if (full_url.split('?')[0] === $(this).attr('href')) {
```
Comment (joelmellon) [inline src/resources/views/base/inc/sidebar.blade.php:65]: Same as above.

**Comment by tabacitu** [comment]
Thanks a lot of for the PR @joelmellon ! I agree that this bit of logic could use some work. It has corner cases like yours that are just not covered well.

I agree with you that splitting by `?` will work well for the Create operation. But my tests show a big problem: **this renders the menu item `inactive` for all _other_ operations**. That's because the sidebar only points to the List operation, not the other ones. That's the entire reason why we chose to go with `startsWith()`, so that all operations for a CRUD make the menu item active.

Screenshot 2020-08-17 at 07 24 10

The current implementation does leave out the use case you mentioned, though, and I hate that. 

**Good news though - I have an out-of-the-box fix for it** 🤣  Something that I've been eyeing for a while now but never got to do because new versions have been feature-packed already: **we should probably make the route segment plural, not singular** (maybe see https://github.com/Laravel-Backpack/Generators/issues/63 for details). We only made it singular to match the Laravel convention at the time, even though we didn't agree to it. Now Laravel has made restful controllers plural, but we haven't, just to keep backwards-compatibility.

But making the route segment plural, we should, in theory, fix this too, because the routes would be:
- `admin/trails`
- `admin/trailheads`

So for _most_ corner cases like you mentioned, it should fix the problem. Right?!

What are your thoughts on this @joelmellon ? Do you have a different _programming_ solution to fix the problem above (sidebar items not active on other operations than List)? If not, I guess a quick fix for _your project_ would be to implement plural route segments already. It's easy, just change the segment name both in `routes/backpack/custom.php` and inside the Controller itself, under `setRoute()`.

Let me know what you think.
Cheers!

**Comment by joelmellon** [comment]
@tabacitu I personally really like the plural URL segment convention, especially for index pages, but I prefer when they're singular for view, and edit, and create. I don't have a strong opinion though, and if Laravel's going in one of these directions, all the more reason to adopt it. Consistency is more important to me.

I don't think it completely solves the problem, but it's a great leap in the short term. The issue I foresee if this logic doesn't change, is in the example I gave for _users/usersettings_. If the `admin/user` route is changed to `admin/users` and there's an `admin/usersettings` route, both would be active in the nav since both start with `admin/users`.

Again though, I do see your proposed solution as an improvement, and I'll be implementing it instead of the fix in the PR here for my own project.

I don't have a ton of time to come up with something better, but if I do I'll let you know. Off the top of my head, I think there nav libraries out there that do this type of thing well, but I couldn't name any. In any case, the first step would be to nail the the definition (aka business rule) for what active means related to the URL. To me it's closely related to what comes before the GET vars, but if a user had GET vars in their nav link, they might expect the menu item to only be active if the exact URL matched, so it may not work for everyone. ‾\\_(ツ)_/‾

**Comment by tabacitu** [comment]
Hmm you're right. There would still be uncovered cases, even with plural. So it still needs better a better JS solution. 
@pxpm what's your take on this?

**Comment by martijnb92** [comment]
I made a small PR https://github.com/Laravel-Backpack/CRUD/pull/3136 to enable the `active` class for menu items in the header. Also works for example the "My account" page in the dropdown user menu.

**Comment by tabacitu** [comment]
> The issue I foresee if this logic doesn't change, is in the example I gave for users/usersettings. If the admin/user route is changed to admin/users and there's an admin/usersettings route, both would be active in the nav since both start with admin/users.

Damn it. You're right @joelmellon there's still that use case that wouldn't be covered...

**Comment by tabacitu** [comment]
@joelmellon looks like we can't really consider this bug fixed. But we can't merge this either. Tough one. I'm going to close the PR though since we can't merge it.

In the future, we'll:
- move towards plural in the route name (should at least _reduce_ the number of issues)
- figure out a different solution for that use case where the plural + second word end up being the same string, for ex:
   - `/admin/users` + `/admin/userprofiles` will be fine;
   - `/admin/users` + `/admin/usersetttings` will NOT be fine, they both have `users`;

But now that I think of it, if the end solution is a better convention (plural) I'm pretty sure another convention will fix the use case above too - using dashes. Right?! Freakin' dashes 🤣  Hear me out - if we use plural for route URLs AND use dashes to split words inside the route URLs we'll end up with:
   - `/admin/users` + `/admin/user-profiles` which will work fine;
   - `/admin/users` + `/admin/user-setttings` which will again work fine;

Quite a simple solution, right?! 🤣 Why didn't we think of this before?!
